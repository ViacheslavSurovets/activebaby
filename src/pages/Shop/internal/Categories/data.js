import { prams, strollers, carSeats, accessories, linens } from '@assets/categories';

export const CATEGORIES_DATA = [
  {
    title: 'categories.strollers',
    imageUrl: strollers,
    linkUrl: 'shop/strollers',
    size: 'big'
  },
  {
    title: 'categories.prams',
    imageUrl: prams,
    linkUrl: 'shop/prams',
    size: 'big'
  },
  {
    title: 'categories.carSeats',
    imageUrl: carSeats,
    linkUrl: 'shop/car-seats',
    size: 'small'
  },
  {
    title: 'categories.accessories',
    imageUrl: accessories,
    linkUrl: 'shop/accessories',
    size: 'small'
  },
  {
    title: 'categories.linens',
    imageUrl: linens,
    linkUrl: 'shop/linens',
    size: 'small'
  }
];
