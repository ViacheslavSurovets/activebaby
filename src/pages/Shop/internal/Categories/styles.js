import styled from 'styled-components';
import { desktop, tablet, mobile } from '@media';

export const CategoryContainer = styled.section`
  display:flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 95vw;
  margin: 0 auto;
`;
