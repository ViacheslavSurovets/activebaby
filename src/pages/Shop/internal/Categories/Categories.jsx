import React from 'react';
import { CategoryItem } from '../CategoryItem';
import { CATEGORIES_DATA } from './data';
import { CategoryContainer } from './styles';

const Categories = ( props ) => (
  <CategoryContainer>
    {
      CATEGORIES_DATA.map ( ( { ...rest }, idx ) =>
        <CategoryItem key={ idx } { ...rest } />
      )
    }
  </CategoryContainer>
);

export default Categories;
