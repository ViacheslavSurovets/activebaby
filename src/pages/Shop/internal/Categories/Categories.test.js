import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Categories from './Categories';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('Categories', ()=> {
  it('Categories Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <Categories  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
