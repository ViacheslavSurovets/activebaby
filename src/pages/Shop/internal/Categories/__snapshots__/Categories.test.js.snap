// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Categories Snapshot 1`] = `ShallowWrapper {}`;
